FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -18,7 +18,63 @@ public final class RemoteFeedLoader: FeedLoader {
 		self.client = client
 	}
 
+	private var OK_200: Int { return 200 }
+
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url) { [weak self] result in
+			guard let self = self else { return }
+			switch result {
+			case let .success((data, httpURLResponse)):
+
+				if httpURLResponse.statusCode != self.OK_200 {
+					completion(.failure(Error.invalidData))
+					return
+				}
+
+				do {
+					let value = try JSONDecoder().decode(RemoteFeedItem.self, from: data)
+					var feedImageList: [FeedImage] = []
+					value.items.forEach { item in
+						feedImageList.append(FeedImage(id: item.id, description: item.description, location: item.location, url: item.url))
+					}
```

Comment:
Instead of using a `forEach` and appending each item, it's possible to use a `map` for this transformation to avoid dealing with a mutable array, and also to make the code more concise and easier to follow.
```suggestion
					let feedImageList = value.items.map { item in
						return FeedImage(id: item.image_id, description: item.image_desc, location: item.image_loc, url: item.image_url)
					}
```